src/main/java/space/obminyashka/items_exchange/controller/UserController.java

```diff
@@ -108,18 +110,15 @@ public String updateUserPassword(@Valid @RequestBody UserChangePasswordDto userC
             @ApiResponse(responseCode = "403", description = "FORBIDDEN"),
             @ApiResponse(responseCode = "409", description = "CONFLICT")})
     @ResponseStatus(HttpStatus.ACCEPTED)
-    public String updateUserEmail(@Valid @RequestBody UserChangeEmailDto userChangeEmailDto, @Parameter(hidden = true) Authentication authentication)
-            throws DataConflictException {
-        User user = getUser(authentication.getName());
-        if (user.getEmail().equals(userChangeEmailDto.getNewEmail())) {
-            throw new DataConflictException(getMessageSource(ResponseMessagesHandler.ExceptionMessage.EMAIL_OLD));
-        }
-        if (userService.existsByEmail(userChangeEmailDto.getNewEmail())) {
-            throw new DataConflictException(getMessageSource(
-                    ResponseMessagesHandler.ValidationMessage.DUPLICATE_EMAIL));
-        }
-
-        return userService.updateUserEmail(userChangeEmailDto, user);
+    public String updateUserEmail(@Parameter(name = "New email")
+                                  @Email(regexp = PatternHandler.EMAIL, message = "{" + ResponseMessagesHandler.ValidationMessage.INVALID_EMAIL + "}")
```

Comment:
You should statically import INVALID_EMAIL and that would be looked a way better.

Reply:
If I statically import INVALID_EMAIL it breaks the build

Reply:
refacted